### DISCUSSION TITLE
Importing is overwriting all fields

### DISCUSSION THREAD

**Discussion by stellarfinn** [Budibase/budibase#11996] [Help] [answered]
I've added a new column to my DB and I am trying to import values for all the rows. Every time I import, it clears the data for all the other fields. 

I'm uploading a CSV with two fields only. A Manufacturer Code (Key) field, and a Storage Condition field (Dry, Cooler, Freezer). 

<img width="1383" alt="image" src="https://github.com/Budibase/budibase/assets/99847728/5c0d59e6-c0c4-4cbb-97a7-ce79b159dc07">

I can successfully import them using the Manufacturer Code as the identifier field but then all existing fields go blank. 

Any thoughts? I don't want to manually go through 245 items to select a Storage condition.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @stellarfinn,

How is the new CSV you are trying to import structured? Does it contain all the columns of the field and data + the new column? With its data populated within it?

If the above is the case we actually do have this functionality available. Below i've provided a screenshot of the sales table, with the new column added.
<img width="1043" alt="Screenshot 2023-10-09 at 11 31 56" src="https://github.com/Budibase/budibase/assets/126772285/70904043-1b86-4d2d-9f3f-2afb0881ef00">

Below is the CSV show a few different rows, with values entered for this new created column within Budibase.
<img width="1319" alt="Screenshot 2023-10-09 at 11 33 09" src="https://github.com/Budibase/budibase/assets/126772285/bfd26ac5-e3fa-4e6b-9aed-17af82257bfc">

Here a gif below on how to use this CSV, to update relevant records with the new column's information.
![Screen Recording 2023-10-09 at 11 37 05](https://github.com/Budibase/budibase/assets/126772285/8b73ac61-e132-4cd2-8d40-a5b2497332b7)

From looking at the information you've provided again it seems that you don't have any data for the other columns, thats why its nulling it. I've created a [feature request](https://github.com/Budibase/budibase/issues/12001), this would allow you to only include the field you wish to match (in my example that would be orderID) + the field you'd want to update (new column in my example.) And prevent nulling of our non included columns.

You may be able to do something with automations (webhooks), if you have the id to match, from there you could format the CSV data into an array loop over the array and update the fields based on their matching records.

I hope this helps.

**Reply by stellarfinn** [reply]
So, to clarify, the first time, I exported all of the rows so I could fill in data for the new column.

Unfortunately, the export included fields which had double quotes in them and the CSV was ruined because of that. Opening the CSV in Excel was a mess. 

So I uploaded a CSV with only two columns, the Identifier and the new column. 

My bigger issue is that the backups did NOT function as one would expect them to. The whole app was destroyed with the backups, some of the data being restored, some not, image links broken, some functioning. I had a very difficult time restoring a backup and then exporting it so that I could import it into a new app. 

I was finally able to accomplish this but it took a backup from three days prior to finally work. This was great and tragic in the same instance since I had made a lot of updates to data in those three days and lost all that work.
